src/awesome/drawin.rs

```diff
@@ -71,15 +97,29 @@ fn method_setup<'lua>(lua: &'lua Lua, builder: ClassBuilder<'lua>) -> rlua::Resu
     // TODO Do properly
     use super::dummy;
     builder.method("connect_signal".into(), lua.create_function(dummy))?
-           .method("__call".into(), lua.create_function(dummy_create))
+           .method("__call".into(), lua.create_function(|lua, _: Value| Drawin::new(lua)))
```

Comment:
Oh and the global (e.g. `drawin`) when called get a table of properties to be applied, for example `visible = true` (but the table can also contains keys that the C code does not understand and thus ignores). This code looks like you explicitly ignore these extra arguments.

Reply:
Nice catch, I'll fix that when I fix the object creation (so I can throw those unknown keys into `data`)

Reply:
I don't think awesome stores unknown keys into `data`, it just really completely ignores them.
This seems to be done in `luaA_class_new` in `common/luaclass.c`: All strings for which it finds a `lua_class_property_t` are handled, all others are just silently ignored.
(This is like this mostly for "historic reasons", I guess; I don't know if it makes sense or not)